rust/src/quic/cyu.rs

```diff
@@ -0,0 +1,165 @@
+use super::{
+    frames::Frame,
+    parser::{QuicHeader, QuicVersion},
+};
+
+#[derive(Debug, PartialEq)]
+pub struct Cyu {
+    pub string: String,
+    pub hash: String,
+}
+
+impl Cyu {
+    pub(crate) fn new(string: String, hash: String) -> Self {
+        Self { string, hash }
+    }
+
+    pub(crate) fn generate(header: &QuicHeader, frames: &[Frame]) -> Option<Cyu> {
+        let version = match header.version {
+            QuicVersion::Q043 => Some("43"),
+            QuicVersion::Q044 => Some("44"),
+            QuicVersion::Q045 => Some("44"),
+            QuicVersion::Q046 => Some("46"),
+            _ => None,
+        }?;
+
+        if frames.is_empty() {
+            return None;
+        }
+
+        for frame in frames {
```

Comment:
I do not know the protocol here.
But this construction feels strange : iterating for frames and returning the first cyu_string
Could not there be a later frame with a different cyu ?
Would not it be a problem ?

Reply:
I assumed there would only be a single STREAM frame per packet (is the case in the testing pcaps) but this may not be the case if multiple connections are multiplexed. I'll make this return a Vec instead.